planner/core/find_best_task.go

```diff
@@ -1591,9 +1591,10 @@ func (is *PhysicalIndexScan) initSchema(idxExprCols []*expression.Column, isDoub
 		}
 	}
 
-	if isDoubleRead {
+	if isDoubleRead || is.Index.Global {
```

Comment:
Does it mean if not an isDoubleRead plan with global index also need extraHandleID here? Could you breif explain it?

Reply:
If IndexScan is global, it should also decode `ExtraPidColID`  from values in TiKV. So that we can filter the rows from needed partitions.
e.g., select id from t partition(p0) use index(global_idx);
In this example, We should tell TiKV to decode `ExtraPidColID` in index value, and check whether we need the row.
If the IndexScan is single read, the `ExtraPidColID` would not return to SQL. Because we don't add its offset in DAG.

Reply:
You are right. For a singleRead, ExtraHandleID is actually useless. 
But unistore/TiKV's decode method has limit for it. I notice if handle is not needed, `decodeIndexKvGeneral` in unisore will return directly and `ExtraPidColID` is not decoded.
![image](https://user-images.githubusercontent.com/20724552/233033727-cb15b521-d0f2-4241-98e8-8f3bd55b13cd.png)